apps/omg/lib/omg/state/core.ex

```diff
@@ -96,34 +128,41 @@ defmodule OMG.State.Core do
         }
 
   @doc """
-  Recovers the ledger's state from data delivered by the `OMG.DB`
+  Initializes the state from the values stored in `OMG.DB`
   """
   @spec extract_initial_state(
-          utxos_query_result :: [list({OMG.DB.utxo_pos_db_t(), OMG.Utxo.t()})],
           height_query_result :: non_neg_integer() | :not_found,
           child_block_interval :: pos_integer()
         ) :: {:ok, t()} | {:error, :top_block_number_not_found}
-  def extract_initial_state(utxos_query_result, height_query_result, child_block_interval)
-      when is_list(utxos_query_result) and is_integer(height_query_result) and is_integer(child_block_interval) do
-    # extract height, last deposit height and utxos from query result
-    height = height_query_result + child_block_interval
-
-    state = %__MODULE__{
-      height: height,
-      utxos: UtxoSet.init(utxos_query_result)
-    }
+  def extract_initial_state(height_query_result, child_block_interval)
+      when is_integer(height_query_result) and is_integer(child_block_interval) do
+    state = %__MODULE__{height: height_query_result + child_block_interval}
 
     {:ok, state}
   end
 
-  def extract_initial_state(
-        _utxos_query_result,
-        :not_found,
-        _child_block_interval
-      ) do
+  def extract_initial_state(:not_found, _child_block_interval) do
     {:error, :top_block_number_not_found}
   end
 
+  @doc """
+  Tell whether utxo position was created or spend by current state.
+  It's an optimization as there is no need to ask DB of known utxos
```

Comment:
drop this line, it is not an optimization, if you don't `or` with `recently_spent` you'll allow double spends.